upi/vsphere/variables.tf

```diff
@@ -1,153 +1,167 @@
+// phpIPAM variables
+
+variable "ipam" {
+  type        = string
+  description = "The IPAM server to use for IP management."
+  default     = ""
+}
+
+variable "ipam_token" {
+  type        = string
+  description = "The IPAM token to use for requests."
+  default     = ""
+}
+
 //////
 // vSphere variables
 //////
 
 variable "vsphere_server" {
-  type        = "string"
+  type        = string
   description = "This is the vSphere server for the environment."
 }
 
 variable "vsphere_user" {
-  type        = "string"
+  type        = string
   description = "vSphere server user for the environment."
 }
 
 variable "vsphere_password" {
-  type        = "string"
+  type        = string
   description = "vSphere server password"
 }
 
 variable "vsphere_cluster" {
-  type        = "string"
+  type        = string
   description = "This is the name of the vSphere cluster."
 }
 
 variable "vsphere_datacenter" {
-  type        = "string"
+  type        = string
   description = "This is the name of the vSphere data center."
 }
 
 variable "vsphere_datastore" {
-  type        = "string"
+  type        = string
   description = "This is the name of the vSphere data store."
 }
 
 variable "vm_template" {
-  type        = "string"
+  type        = string
   description = "This is the name of the VM template to clone."
 }
 
 variable "vm_network" {
-  type        = "string"
+  type        = string
   description = "This is the name of the publicly accessible network for cluster ingress and access."
   default     = "VM Network"
 }
 
-variable "ipam" {
-  type        = "string"
-  description = "The IPAM server to use for IP management."
-  default     = ""
-}
-
-variable "ipam_token" {
-  type        = "string"
-  description = "The IPAM token to use for requests."
-  default     = ""
+variable "vm_dns_addresses" {
+  type    = list(string)
+  default = ["1.1.1.1", "9.9.9.9"]
 }
 
 /////////
 // OpenShift cluster variables
 /////////
 
 variable "cluster_id" {
-  type        = "string"
+  type        = string
   description = "This cluster id must be of max length 27 and must have only alphanumeric or hyphen characters."
 }
 
 variable "base_domain" {
-  type        = "string"
+  type        = string
   description = "The base DNS zone to add the sub zone to."
 }
 
 variable "cluster_domain" {
-  type        = "string"
+  type        = string
   description = "The base DNS zone to add the sub zone to."
 }
 
 variable "machine_cidr" {
-  type = "string"
+  type = string
 }
 
 /////////
 // Bootstrap machine variables
 /////////
 
-variable "bootstrap_complete" {
-  type    = "string"
-  default = "false"
+variable "bootstrap_ignition_path" {
+  type    = string
+  default = "./bootstrap.ign"
 }
 
-variable "bootstrap_ignition_url" {
-  type = "string"
+variable "bootstrap_complete" {
+  type    = string
+  default = "false"
 }
 
-variable "bootstrap_ip" {
-  type    = "string"
-  default = ""
+variable "bootstrap_ip_address" {
```

Comment:
why is this `ip_address` but a list?

Reply:
the ipam module takes a list.  Trying to reuse as much as I can

Reply:
if this is a sticking point, you can just initialize a list with the variable when passing to module. so if you switch this to string then the input to ipam module would be `addresses = [var.bootstrap_ip_address]`